cluster.c

```diff
@@ -783,7 +781,7 @@ RRStatus ShardingInfoValidateShardGroup(RedisRaftCtx *rr, ShardGroup *new_sg)
 /* Update an existing ShardGroup in the active ShardingInfo.
  *
  * FIXME: We currently only handle updating nodes but don't support remapping
- *        hash slots.
+ *        hash slots, via this mechanism
```

Comment:
in practice, this is only used for the internal gossipy sharding mechanism, where the thread can update individual shardgroups with an updated set of nodes, but isn't about replacing the whole set of shard groups as one gets with an external orchestrator.

The question becomes, should the internal mechanism be moved to replace (it would have to construct a whole list)

Reply:
For now, this is useful for testing if not anything else. In the future it could become irrelevant.